--- original	2025-03-05 20:47:34
+++ patched	2025-03-05 20:47:34
@@ -20,13 +20,22 @@
 external create_parser :
   unit -> Tree_sitter_API.ts_parser = "octs_create_parser_"
 
-let ts_parser = create_parser ()
+(* NOTE: Ok because we run one target per domain at any 1 time. *)
+let ts_parser = Domain.DLS.new_key create_parser
 
 let parse_source_string ?src_file contents =
-  Tree_sitter_parsing.parse_source_string ?src_file ts_parser contents
+  Fun.protect ~finally:Fun.id (* reset_parser *)
+    (fun () ->
+      Tree_sitter_parsing.parse_source_string ?src_file
+        (Domain.DLS.get ts_parser)
+        contents)
 
 let parse_source_file src_file =
-  Tree_sitter_parsing.parse_source_file ts_parser src_file
+  Fun.protect ~finally:Fun.id (* reset_parser *)
+    (fun () ->
+      Tree_sitter_parsing.parse_source_file
+        (Domain.DLS.get ts_parser)
+        src_file)
 
 let extras = [
   "comment";
